### PR TITLE
FCREPO-2632. Use 'acl:agentClass foaf:Agent' to denote public access

### PR DESCRIPTION
**Pull request by mohideen**
Resolves: https://jira.duraspace.org/browse/FCREPO-2632

An alternate implementation of PR: #1321 

# What does this Pull Request do?
Changes to treat foaf:Agent as an acl:agentClass instead of acl:agent.

# What's new?
The getRoles method in the roles provider class will also include agentClass roles for the foaf:Agent.

# How should this be tested?

Maven build with tests.

# Interested parties
@whikloj

### CODE REVIEW

File: fcrepo-auth-webac/src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java [hunk 1]
```diff
@@ -244,10 +245,16 @@ private FedoraResource getBaseVersion(final FedoraResource resource) {
...
+                    .filter(agent -> !agent.equals(FOAF_AGENT_VALUE))
                     .forEach(agent -> {
                         effectiveRoles.computeIfAbsent(agent, key -> new HashSet<>())
                             .addAll(auth.getModes().stream().map(URI::toString).collect(toSet()));
                     });
+                auth.getAgentClasses().stream().filter(agentClass -> agentClass.equals(FOAF_AGENT_VALUE))
+                    .forEach(agentClass -> {
+                        effectiveRoles.computeIfAbsent(agentClass, key -> new HashSet<>())
+                            .addAll(auth.getModes().stream().map(URI::toString).collect(toSet()));
+                    });
```
Comment (whikloj) [inline fcrepo-auth-webac/src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java]: @mohideen it looks like you filter out `FOAF_AGENT_VALUE` from `effectiveRoles` in line 248 and then re-add them here, is that correct?

Comment (whikloj) [reply fcrepo-auth-webac/src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java]: Oh sorry I missed the change from `agent` to `agentClass`.

Comment (mohideen) [reply fcrepo-auth-webac/src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java]: @whikloj  Yes, I am filtering out FOAF_AGENT_VALUE from the agents list and allowing only FOAF_AGENT_VALUE in the agentClasses list of the auth.

**Comment by whikloj** [comment]
@mohideen nice work. 

Can I ask that you add one test that checks that the old `acl:agent foaf:Agent` authorization doesn't continue to work with this and any future changes.

**Comment by mohideen** [comment]
@whikloj Added the test.

**Pull request merged**